Add more stubs for redis-py

brainix opened the pull request:
I've tested these stubs against my own project here:
https://github.com/brainix/pottery/pull/240

The output is clean for all these new stubs when running `$ mypy
--strict`.

srittau commented on third_party/2and3/redis/client.pyi:
```diff
@@ -221,7 +221,7 @@ class Redis(object):
     def smove(self, src, dst, value): ...
     def spop(self, name, count: Optional[int] = ...): ...
-    def srandmember(self, name, number=...): ...
+    def srandmember(self, name: _Key, number: Optional[int] = ...) -> _Value: ...
```
This returns `Optional[_Value]` if `number` is not given or is `None` and a list of values if `number` is not `None`. We should use overloads to reflect this.

srittau commented on third_party/2and3/redis/client.pyi:
```diff
@@ -159,12 +159,12 @@ class Redis(object):
     def __setitem__(self, name, value): ...
     def setbit(self, name: _Key, offset: int, value: int) -> int: ...
-    def setex(self, name, time, value): ...
+    def setex(self, name: _Key, time: Union[int, timedelta], value: _Value): ...
```
Returns `bool`.

srittau reviewed (changes requested):
Thanks, two remarks below.

brainix commented:
Hi, @srittau.

Thanks for the feedback; I believe I've addressed your concerns. However, now, whenever I run `$ mypy --strict`, one of the errors that I get is:

brainix commented:
Thanks for the feedback, @srittau. I believe that I've addressed your concerns. How does this Pull Request look now?

srittau commented on third_party/2and3/redis/client.pyi:
```diff
@@ -221,7 +221,10 @@ class Redis(object):
     def smove(self, src, dst, value): ...
     def spop(self, name, count: Optional[int] = ...): ...
-    def srandmember(self, name, number=...): ...
+    @overload
+    def srandmember(self, name: _Key, number: None) -> Optional[_Value]: ...
+    @overload
+    def srandmember(self, name: _Key, number: int) -> List[_Value]: ...
```
One little thing and then this is ready to go. The first overload needs `number: None = ...` as `srandmember()` has a default (`None`) for the second argument.
